src/grt/grt-wave_options-read.adb

```diff
@@ -0,0 +1,125 @@
+   begin
+      return Find_Cursor (Name, Trees (Index));
+   end Get_Top_Cursor;
+
+   function Get_Cursor (Name : Ghdl_C_String; Parent : Elem_Acc;
+            Is_Signal : Boolean := False) return Elem_Acc is
+   begin
+      if Display_All_Signals then
+         return null;
+      end if;
+      return Find_Cursor (Name, Parent.Next_Child, Is_Signal);
+   end Get_Cursor;
+
+   function Is_Displayed (Cursor : Elem_Acc) return Boolean is
+   begin
+      if Display_All_Signals or else Cursor /= null then
+         return True;
+      end if;
+      return False;
+   end Is_Displayed;
+
+   procedure Check_If_All_Found is
+   begin
+      for Index in Tree_Index_Type'Range loop
+         Check_If_Found (Trees (Index), Seps (Index), 1);
+      end loop;
+   end Check_If_All_Found;
+
+-- private --------------------------------------------------------------------
+
+   function Find_Cursor (Name : Ghdl_C_String; First : Elem_Acc;
+            Is_Signal : Boolean := False) return Elem_Acc is
+      Len : constant Natural := strlen (Name);
+      Cursor : Elem_Acc := First;
+   begin
+      loop
+         if Cursor = null then
+            return null;
+         elsif Cursor.Name.all = Name (1 .. Len) then
+            if Is_Signal then
+               Cursor.Kind := Signal;
+            else
+               Cursor.Kind := Pkg_Entity;
+            end if;
+            return Cursor;
+         end if;
+         Cursor := Cursor.Next_Sibling;
+      end loop;
+   end Find_Cursor;
+
+   procedure Check_If_Found (Previous_Cursor : Elem_Acc;
+                             Sep : Character; Level : Positive) is
+      Cursor : Elem_Acc := Previous_Cursor;
+      Index : Positive;
+   begin
+      while Cursor /= null loop
+         if Cursor.Kind = Not_Found then
+            Print_Context (Cursor.Line_Context, Warning);
+            Report_C ("no VHDL object in design matches ");
+            -- Display the path of the first unfound vhdl object in signal path
+            if Level > 1 then
+               Index := Cursor.Line_Context.Str'First;
+               for I in 2 .. Level loop
+                  Index := Find (Cursor.Line_Context.Str.all, Sep, Index + 1);
+               end loop;
+               Report_C (Cursor.Line_Context.Str (Cursor.Line_Context.Str'First
+                                                  .. Index));
+            elsif Sep = '/' then
+               Report_C ("/");
+            end if;
+            Report_E (Cursor.Name.all);
+         elsif Level = Cursor.Line_Context.Max_Level
+           and then Cursor.Kind = Pkg_Entity
+         then
+            Error_Context ("not a signal", Cursor.Line_Context, Warning);
+         else
+            Check_If_Found (Cursor.Next_Child, Sep, Level + 1);
+         end if;
+         Cursor := Cursor.Next_Sibling;
+      end loop;
+
+   end Check_If_Found;
+
+   function Display_All_Signals return Boolean is
+   begin
+      if Trees = Tree_Array'(others => null) then
```

Comment:
Fixed but gnat didn't allow me to write it that way (is it a compiler 
bug ?). I had to write instead :

return Trees = Tree_Array'(others => null);


Le 23. 07. 16 à 12:26, tgingold a écrit :
>
> In src/grt/grt-wave_options-read.adb 
> <https://github.com/tgingold/ghdl/pull/121#discussion_r71970740>:
>
> > +            Report_E (Cursor.Name.all);
> > +         elsif Level = Cursor.Line_Context.Max_Level
> > +           and then Cursor.Kind = Pkg_Entity
> > +         then
> > +            Error_Context ("not a signal", Cursor.Line_Context, Warning);
> > +         else
> > +            Check_If_Found (Cursor.Next_Child, Sep, Level + 1);
> > +         end if;
> > +         Cursor := Cursor.Next_Sibling;
> > +      end loop;
> > +
> > +   end Check_If_Found;
> > +
> > +   function Display_All_Signals return Boolean is
> > +   begin
> > +      if Trees = Tree_Array'(others => null) then
>
> You can write:
> return Trees = (others => null);
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/tgingold/ghdl/pull/121/files/4bf3293563ba8e95cc35992d029ff26c4f4e67e5#r71970740>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/ATNePli1cH3REQRepjRqY37u_ufAi5Xcks5qYexWgaJpZM4JTTAB>.
>